### ISSUE TITLE
[v2] Reached invalid decision id 0 while looking through

### ISSUE THREAD

**Issue by williamdes** [composer/composer#9012]
My `composer.json`:

```json
{
    "require": {
        "php": ">=7.1.7",
        "astrotomic/laravel-translatable": "^11.7",
        "barryvdh/laravel-snappy": "^0.4.4",
        "brozot/laravel-fcm": "^1.2",
        "daltcore/lara-pdf-merger": "^2.0",
        "doctrine/dbal": "^2.9",
        "eluceo/ical": "^0.16.0",
        "ezyang/htmlpurifier": "^4.9",
        "facebook/graph-sdk": "^5.7",
        "biscolab/laravel-recaptcha": "^4.0",
        "guzzlehttp/guzzle": "~6.3",
        "intervention/image": "^2.4",
        "intervention/validation": "^2.4",
        "jenssegers/agent": "^2.6",
        "jenssegers/date": "^4.0",
        "laravel/framework": "^6.0",
        "laravel/helpers": "^1.1.1",
        "laravel/tinker": "^2.4",
        "laravelcollective/html": "^6.0",
        "league/flysystem-aws-s3-v3": "^1.0",
        "league/fractal": "^0.17.0",
        "maatwebsite/excel": "^3.1",
        "mattketmo/email-checker": "^1.5",
        "omniphx/forrest": "^2.6",
        "pixelpeter/laravel5-isocodes-validation": "^3.0",
        "predis/predis": "^1.1",
        "propaganistas/laravel-phone": "^4.0",
        "ramsey/uuid": "^3.7",
        "sentry/sentry-laravel": "^1.0",
        "symfony/intl": "^4.0",
        "therezor/laravel-transactional-jobs": "^1.0",
        "tymon/jwt-auth": "1.0.0",
        "ext-json": "*"
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.2",
        "barryvdh/laravel-ide-helper": "^2.6",
        "editorconfig-checker/editorconfig-checker": "^10.0",
        "filp/whoops": "^2.3",
        "fzaninotto/faker": "^1.8",
        "johnkary/phpunit-speedtrap": "^3.2",
        "mockery/mockery": "^1.2",
        "nunomaduro/collision": "^4.0",
        "nunomaduro/larastan": "^0.6",
        "phpunit/phpunit": "^8.0",
        "slevomat/coding-standard": "^6.1",
        "squizlabs/php_codesniffer": "^3.3",
        "swaggest/json-schema": "^0.12.6",
        "zircote/swagger-php": "^3.0"
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true,
        "process-timeout": 0
    }
}

```

Output of `composer diagnose`:

```
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 2.0.0-alpha1
PHP version: 7.2.31
PHP binary path: /usr/local/bin/php
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

When I run this command:

```
composer update
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies

                                                                                                                                                                                                                  
  [Composer\DependencyResolver\SolverBugException]                                                                                                                                                                
  Reached invalid decision id 0 while looking through (131|132|133|134|135|136|137|138|139|140|141|142|143|144|145|146|147|148|149|150|151|152|153|154|155|156|157|158|159|160|161|162|163|164|165|166|167|168|1  
  69|170|171|172|173|174|175|176|177|178|179|180|181|182|183) for a literal present in the analyzed rule (-131|2513|2514|2515|2516|2517|2518|2519|2520|2521|2522|2523|2524|2525|2526|2527|2528|2529|2530|2531).   
  This exception was most likely caused by a bug in Composer.                                                                                                                                                     
  Please report the command you ran, the exact error you received, and your composer.json on https://github.com/composer/composer/issues - thank you!                                                             
                                                                                                                                                                                                                  

update [--with WITH] [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-autoloader] [--no-scripts] [--no-suggest] [--no-progress] [--with-dependencies] [--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...

```

And I expected this to happen:

Do the update

NOTE: **it works before applying this diff** 
```diff
diff --git a/composer.json b/composer.json
index 79771ded5..edb4521e1 100644
--- a/composer.json
+++ b/composer.json
@@ -46,7 +46,7 @@
         "fzaninotto/faker": "^1.8",
         "johnkary/phpunit-speedtrap": "^3.2",
         "mockery/mockery": "^1.2",
-        "nunomaduro/collision": "^3.0",
+        "nunomaduro/collision": "^4.0",
         "nunomaduro/larastan": "^0.6",
         "phpunit/phpunit": "^8.0",
         "slevomat/coding-standard": "^6.1",
```

**Comment by aios** [comment]
+1 confirm

`{
    "name": "laravel/laravel",
    "type": "project",
    "description": "The Laravel Framework.",
    "keywords": [
        "framework",
        "laravel"
    ],
    "license": "MIT",
    "require": {
        "php": "^7.4",
        "ext-json": "*",
        "ext-mbstring": "*",
        "botble/media": "*@dev",
        "butschster/meta-tags": "^1.4",
        "davejamesmiller/laravel-breadcrumbs": "5.x",
        "dionisiy13/google-news-sitemap": "^1.0",
        "fideloper/proxy": "^4.0",
        "facade/ignition": "^2.0",
        "fruitcake/laravel-cors": "^2.0",
        "fzaninotto/faker": "^1.8",
        "guzzlehttp/guzzle": "^7.0.1",
        "hemp/presenter": "^2.0",
        "ihorhnatchuk/laravel-browsershot": "dev-master",
        "intervention/image": "^2.5",
        "laracasts/flash": "^3.0",
        "laravel/framework": "^7.0",
        "laravel/socialite": "^4.1",
        "laravel/tinker": "^2.0",
        "laravel/ui" : "^2.0",
        "maatwebsite/excel": "^3.1",
        "predis/predis": "^1.1",
        "socialiteproviders/instagram": "^3.0",
        "socialiteproviders/odnoklassniki": "^1.0",
        "socialiteproviders/vkontakte": "^4.0",
        "spatie/laravel-feed": "^2.7",
        "spatie/laravel-sitemap": "^5.7",
        "spatie/laravel-sluggable": "^2.1",

        "webwizo/laravel-shortcodes": "1.0.*"
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.2",
        "beyondcode/laravel-dump-server": "^1.0",
        "filp/whoops": "^2.0",
        "laravel/telescope": "^3.0",
        "mockery/mockery": "^1.0",
        "nunomaduro/collision": "^4.2.0",
        "phpunit/phpunit": "^8.5"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "repositories": [{
            "type": "path",
            "url": "./media"
        }
    ],
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/"
        },
        "classmap": [
            "database/seeds",
            "database/factories"
        ],
        "files": [
            "app/helpers.php",
            "app/macros.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "scripts": {
        "pre-operations-exec": [
            "@php artisan key:generate --ansi"
        ],
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ]

    }
}
`
```composer update``` - taking 

`
...
Generating rules
Resolving dependencies through SAT
Looking at all rules.
Something's changed, looking at all rules again (pass #652)

  [Composer\DependencyResolver\SolverBugException]
  Reached invalid decision id 0 while looking through (-14902) for a literal present in the analyzed rule (-388|8047|8048|8049|8050|8051|8052|8053|8054|8055|8056|8057|8058|8059|8060|8061|8062|8063|8064|8065|8
  066|8067|8068|8069|8070|8071|8072|8073|8074|8075|8076).
  This exception was most likely caused by a bug in Composer.
  Please report the command you ran, the exact error you received, and your composer.json on https://github.com/composer/composer/issues - thank you!


Exception trace:
 () at phar:///usr/local/bin/composer/src/Composer/DependencyResolver/Solver.php:432
 Composer\DependencyResolver\Solver->analyze() at phar:///usr/local/bin/composer/src/Composer/DependencyResolver/Solver.php:316
 Composer\DependencyResolver\Solver->setPropagateLearn() at phar:///usr/local/bin/composer/src/Composer/DependencyResolver/Solver.php:364
 Composer\DependencyResolver\Solver->selectAndInstall() at phar:///usr/local/bin/composer/src/Composer/DependencyResolver/Solver.php:775
 Composer\DependencyResolver\Solver->runSat() at phar:///usr/local/bin/composer/src/Composer/DependencyResolver/Solver.php:210
 Composer\DependencyResolver\Solver->solve() at phar:///usr/local/bin/composer/src/Composer/Installer.php:411
 Composer\Installer->doUpdate() at phar:///usr/local/bin/composer/src/Composer/Installer.php:249
 Composer\Installer->run() at phar:///usr/local/bin/composer/src/Composer/Command/UpdateCommand.php:228
 Composer\Command\UpdateCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:245
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:835
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:185
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:294
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:120
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:61
 require() at /usr/local/bin/composer:24

update [--with WITH] [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-autoloader] [--no-scripts] [--no-suggest] [--no-progress] [--with-dependencies] [--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...
`

**Comment by aios** [comment]
@Seldaek Maybe need to make some details on logs. I have proposal to add "Which package call that state"

**Comment by aios** [comment]
@naderman @Seldaek May be it can take some more info.
It causes only when xdebug extensions is `on`

**Comment by anouarabdsslm** [comment]
I just get this issue today. @Seldaek  how did u solve this ?

**Comment by Seldaek** [comment]
We have not solved it. AFAIK it only happens when there are issues resolving dependencies.. But you can try using Composer 1 to see what the problem is perhaps, or see if it works there then please report details.

**Comment by aios** [comment]
@Seldaek how we can check the package or group packages wich causes it?

**Comment by Seldaek** [comment]
@aios as I said you can try with v1 which might give you more info. With v2 there isn't really a way until we fix the bug.

**Comment by Slamdunk** [comment]
Confirm:

1. Source code: https://github.com/Slamdunk/storageless/tree/ca88182a71c23945dbbcdc264c7ee7975f456ea9
1. Failing build: https://github.com/Slamdunk/storageless/runs/1155304146

Downgrading to v1 fixes the issue

**Comment by jasorlando** [comment]
:)

**Comment by naderman** [comment]
Should be fixed in https://github.com/composer/composer/pull/9312 - can anyone here verify that fixes the issue?

**Comment by Seldaek** [comment]
This is solved by #9312 - it now returns a fairly long error output and still takes 196seconds of solving time, but the most important ones boil down to:

```
    - laravel/framework v6.0.0 requires symfony/console ^4.3.4
    - nunomaduro/collision v4.0.0 requires symfony/console ^5.0 
```

Which shows the conflict and why it can not find a solution here.
